src/cpr_sdk/ssm.py

```diff
@@ -0,0 +1,34 @@
+import boto3
+
+from cpr_sdk.search_adaptors import VespaSearchAdapter
+
+
+def get_aws_ssm_param(param_name: str, region_name: str = "eu-west-1") -> str:
+    """Retrieve a parameter from AWS SSM"""
+    ssm = boto3.client("ssm", region_name=region_name)
+    response = ssm.get_parameter(Name=param_name, WithDecryption=True)
+    return response["Parameter"]["Value"]
+
+
+def get_vespa_search_adapter_from_aws_secrets(
+    cert_dir: str,
+    vespa_instance_url_param_name: str = "VESPA_INSTANCE_URL",
+    vespa_public_cert_param_name: str = "VESPA_PUBLIC_CERT",
+    vespa_private_key_param_name: str = "VESPA_PRIVATE_KEY",
+) -> VespaSearchAdapter:
+    """
+    Get a VespaSearchAdapter instance by retrieving secrets from AWS Secrets Manager.
+
+    We then save the secrets to local files in the cert_dir directory and instantiate
+    the VespaSearchAdapter.
```

Comment:
I'm uneasy about this. On the one hand I can see this being great for programmatic access, we can stop passing in these certs env variables. Putting this in the sdk though makes it seem like its an option for anyone to use. Leading to potentially any dev having write access to prod as standard. which none of us need or should have.

Reply:
I hear your concern, someone could assume their prod aws profile and call this function and have access.

We're currently storing secrets in systems manager, this doesn't allow you to provide access permissions on the secret object itself. We'd have to limit access at the iam role level (admin and power user profiles). 

Therefore, I'm going to suggest we take a look at using AWS Secrets instead, this allows us to set permissions on the secret object itself. So we can enforce something like "only this service can access this secret" to block access.

Reply:
We've spoken offline, decided to hold off on merging this function until we have secrets with RBAC.